pkg/autoscaler/autoscaler.go

```diff
@@ -877,4 +895,130 @@ var _ = Describe("Autoscaler should", framework.LabelAutoscaler, Serial, func()
+				cleanupObjects[asr.GetName()] = asr
+			}
+
+			By("Create cluster-autoscaler-priority-expander")
+			priorities := "10: \n  - .*" + transientMachineSets[0].GetName() + ".*\n20:\n  - .*" + transientMachineSets[1].GetName() + ".*"
+			cm := configMapResource(priorities)
+			Expect(client.Create(ctx, cm)).Should(Succeed(), "Failed to create ConfigMap with priorities %s", priorities)
+			cleanupObjects[cm.GetName()] = cm
+
+			jobReplicas := int32(4)
+			uniqueJobName := fmt.Sprintf("%s-priority-expander", workloadJobName)
+			By(fmt.Sprintf("Creating scale-out workload %s: jobs: %v, memory: %s",
+				uniqueJobName, jobReplicas, workloadMemRequest.String()))
+			workload := framework.NewWorkLoad(jobReplicas, workloadMemRequest, uniqueJobName, autoscalingTestLabel, "", corev1.NodeSelectorRequirement{
+				Key:      targetedNodeLabel,
+				Operator: corev1.NodeSelectorOpExists,
+			})
+			cleanupObjects[workload.GetName()] = workload
+			Expect(client.Create(ctx, workload)).Should(Succeed(), "Failed to create scale-out workload %s", uniqueJobName)
+
+			By("Check machinesets scale up to 2 machines")
+			expectedReplicas := int32(2)
+			Eventually(func() bool {
+				ms, err := framework.GetMachineSet(ctx, client, transientMachineSets[0].GetName())
+				Expect(err).ToNot(HaveOccurred(), "Failed to get MachineSets %s", transientMachineSets[0].GetName())
+				ms1, err := framework.GetMachineSet(ctx, client, transientMachineSets[1].GetName())
+				Expect(err).ToNot(HaveOccurred(), "Failed to get MachineSets %s", transientMachineSets[1].GetName())
+
+				By(fmt.Sprintf("Waiting for machineSets replicas to scale out. Current replicas for machineset %s are %v, expected %v, for machineset %s are %v, expected %v .",
+					transientMachineSets[0].GetName(), *ms.Spec.Replicas, expectedReplicas, transientMachineSets[1].GetName(), *ms1.Spec.Replicas, expectedReplicas))
+
+				return *ms.Spec.Replicas == expectedReplicas && *ms1.Spec.Replicas == expectedReplicas
+			}, framework.WaitMedium, pollingInterval).Should(BeTrue(), "MachineSet %s failed to scale out to %d replicas or MachineSet %s failed to scale out to %d replicas", transientMachineSets[0].GetName(), expectedReplicas, transientMachineSets[1].GetName(), expectedReplicas)
+
+			By("Check autoscaler scales up based on Priority")
+			Eventually(func() bool {
+				machineP10, err := framework.GetLatestMachineFromMachineSet(ctx, client, transientMachineSets[0])
+				Expect(err).ToNot(HaveOccurred(), "Failed to get the last provisioned machine %s", machineP10)
+				machineP20, err := framework.GetLatestMachineFromMachineSet(ctx, client, transientMachineSets[1])
+				Expect(err).ToNot(HaveOccurred(), "Failed to get the last provisioned machine %s", machineP20)
+
+				return CompareMachineCreationTime(machineP10, machineP20)
+			}, framework.WaitMedium, pollingInterval).Should(BeTrue(), "P20 machineset has high priority and should be scaled up first.")
```

Comment:
When running an `Eventually` assertions, if you wish to use gomega assertions within the function, you should declare and use a local gomega
```suggestion
			Eventually(func(g Gomega) bool {
				machineP10, err := framework.GetLatestMachineFromMachineSet(ctx, client, transientMachineSets[0])
				g.Expect(err).ToNot(HaveOccurred(), "Failed to get the last provisioned machine %s", machineP10)
				machineP20, err := framework.GetLatestMachineFromMachineSet(ctx, client, transientMachineSets[1])
				g.Expect(err).ToNot(HaveOccurred(), "Failed to get the last provisioned machine %s", machineP20)

				return CompareMachineCreationTime(machineP10, machineP20)
			}, framework.WaitMedium, pollingInterval).Should(BeTrue(), "P20 machineset has high priority and should be scaled up first.")
```